module/zfs/dsl_dir.c

```diff
@@ -623,9 +614,7 @@ dsl_dir_init_fs_ss_count(dsl_dir_t *dd, dmu_tx_t *tx)
 	for (zap_cursor_init(zc, os, dsl_dataset_phys(ds)->ds_snapnames_zapobj);
 	    zap_cursor_retrieve(zc, za) == 0;
 	    zap_cursor_advance(zc)) {
-		/* Don't count temporary snapshots */
-		if (za->za_name[0] != '%')
-			my_ss_cnt++;
+		my_ss_cnt++;
```

Comment:
I don't think that we need to change the snapshot count handling to fix this bug.  Should we leave it alone to avoid the possibility of messing up any existing accounting?